### ISSUE TITLE
Multi Project Build fails on gradle clean?

### ISSUE THREAD

**Issue by confile** [j2objc-contrib/j2objc-gradle#252]
I have projectB which includes projectA. I need to translate the code if projectA as well as the code of projectB. 

- Why must projectA also use j2objc plugin?

ProjectA has the following in its build gradle: 

```
apply plugin: 'com.github.j2objccontrib.j2objcgradle'
buildscript {
    dependencies {
    classpath files('/Users/mg/Downloads/j2objc-gradle/build/libs/j2objc-gradle-0.2.2-alpha.jar')
  }
}
```

Doing ``gradle clean`` on projectB leaves to the following error: 

```
A problem occurred evaluating root project 'projectB'.
> project ':projectA' does not use the j2objc plugin.
  dependsOnJ2objcLib can be used only with another project that
  itself uses the j2objc plugin.
```

**Comment by confile** [comment]
@advayDev1 I suppose you could help me with that?

**Comment by advayDev1** [comment]
@confile - I don't understand this phrase: ` I need to translate the code if projectA...`

You also said `ProjectA has the following in its build` - did you mean ProjectB?
Both projectA and projectB need the j2objc plugin applied, since projectB's objective c will reference projectA's objective c code.

**Comment by confile** [comment]
@advayDev1 I mean I need to translate the code of projectA and projectB to Objective-c

**Comment by confile** [comment]
@advayDev1 Let me rephrase this: 
ProjectB includes projectA. 

project B's build gradle is: 

```
apply plugin: 'java'
apply plugin: 'apt'
apply plugin: 'eclipse'
apply plugin: 'com.github.j2objccontrib.j2objcgradle'

sourceCompatibility = 1.7
version = '1.0'

eclipse {
  classpath {
     downloadSources=true
     downloadJavadoc=true
  }
}

buildscript {
  repositories {
    mavenCentral()
    jcenter()
    maven {
      url "https://oss.sonatype.org/content/repositories/snapshots/"
    }
    maven {
      url uri('../../0Gradle-Repo')
    }
  }
  dependencies {
    classpath 'com.jimdo.gradle:gradle-apt-plugin:0.5-SNAPSHOT'
   
    classpath files('/Users/mg/Downloads/j2objc-gradle/build/libs/j2objc-gradle-0.2.2-alpha.jar')
  }
}

 
repositories {
  mavenCentral()
  maven {
    name = "sonatype"
    url = "https://oss.sonatype.org/content/repositories/snapshots/"
  }
  maven {
    name = "sonatypeGoogle"
    url = "https://oss.sonatype.org/content/repositories/google-snapshots/"
  }
  flatDir {
    dirs 'lib'
  }
}

sourceSets {
  apt{
    java{
      srcDir 'build/source/apt'
      srcDir 'build/source/base'
    }
  }
  main {
    java {
      srcDir 'build/source/apt'
      srcDir 'build/source/base'
    }
  }
}
dependencies {
  apt (project(':projectA'))
  apt 'com.google.auto.factory:auto-factory:1.0-beta2'
  apt 'com.google.dagger:dagger:2.0.1'
  apt 'com.google.dagger:dagger-producers:2.0-beta'
  apt 'com.google.dagger:dagger-compiler:2.0.1'

  compile (project(':projectA'))
  compile 'com.google.guava:guava:14.0.1'
  compile 'javax.inject:javax.inject:1'
  compile 'org.json:json:20140107'
  
  compile 'com.google.auto.factory:auto-factory:1.0-beta2'
  compile 'com.google.auto.service:auto-service:1.0-rc2'
  
  compile name: 'j2objc_annotations'
  compile 'com.google.dagger:dagger:2.0.1'
}

j2objcConfig {
 
  dependsOnJ2objcLib ":projectA"
  
  generatedSourceDirs "build/source/apt"

  translateArgs "--no-package-directories"
  translateArgs "--prefixes", "prefixes.properties"
  translateArgs "-use-arc"
  translateArgs "--doc-comments"
  translateArgs "--build-closure"
  translateArgs "--generate-deprecated"

  translateClassPaths "lib/json-20140107.jar", "lib/javax.inject-1.jar", "lib/auto-factory-1.0-beta2.jar", "lib/auto-service-1.0-rc2.jar", "lib/javawriter-2.5.1.jar", "lib/guava-18.0.jar"
    translateSourcepaths "${projectDir}/libSrc/dagger-2.0.1-sources.jar"
 
  extraObjcCompilerArgs '-fobjc-arc'
  finalConfigure()
}
```

project A's build gradle is: 

```
apply plugin: 'java'
apply plugin: 'eclipse'
apply plugin: 'com.github.j2objccontrib.j2objcgradle'
apply plugin: 'com.github.j2objccontrib.j2objcgradle'
buildscript {
    dependencies {
    classpath files('/Users/mg/Downloads/j2objc-gradle/build/libs/j2objc-gradle-0.2.2-alpha.jar')
  }
}
j2objcConfig {
  finalConfigure()
}
```

Doing gradle clean on projectB leaves to the following error:

```
A problem occurred evaluating root project 'projectB'.
> project ':projectA' does not use the j2objc plugin.
  dependsOnJ2objcLib can be used only with another project that
  itself uses the j2objc plugin.
```

**Comment by confile** [comment]
@advayDev1 do you have any idea what is wrong here?

**Comment by advayDev1** [comment]
not sure yet.  one please eliminate the duplicate plugin application:
`
apply plugin: 'com.github.j2objccontrib.j2objcgradle'
apply plugin: 'com.github.j2objccontrib.j2objcgradle'
`
and only keep one.

then, run `./gradlew build --no-daemon --stacktrace`
and give the entire error with stacktrace please.

**Comment by advayDev1** [comment]
also show the associated settings.gradle and build.gradle used in the root project.  feel free to store them as gists instead of copy pasting

**Comment by advayDev1** [comment]
also: indicate what version of the plugin you are using.  Which commit did you build the plugin at?

**Comment by confile** [comment]
@advayDev1 I always use the latest upstream master. This time I use commit: 556edca1ccbaaaf8951a1e78b935175a6064c080 Subject:	Merge pull request #253 from j2objc-contrib/release_v0.3.0-alpha

**Comment by confile** [comment]
@advayDev1I set in my config: 

```
  translateJ2objcLibs "j2objc_annotations.jar", "j2objc_guava.jar", "javax.inject-1.jar", "jsr305-3.0.0.jar"
```

and got the error: 

```
> Could not find method translateJ2objcLibs() for arguments [j2objc_annotations.jar, j2objc_guava.jar, javax.inject-1.jar, jsr305-3.0.0.jar] on root project 'majestella-main'.
```

Without the above line clean works with the latest master as referenced in my previous statement.

When running ``gradle j2ojbcXcode`` I get the following error: 

```
:projectA:j2objcTranslate
Error during translation:
error: /Users/mg/Documents/Grails/GGTS3.6.3-SR1-2/projectA/src/main/java/projectA/core/prototype/processor/proxy/AnnotatedProxyAbstract.java:7: The import javax.lang.model.element.Element cannot be resolved
``` 

I know what the problem is. When I did my copy task: 

```
task copyBaseSources(type: Copy) {
  description "Copy all source files from majestella-base into the current project."
  exclude '**/*processor*'  // Annotation Processors should not be copied and translated
 from ...
 into ...
}
```

I excluded the annotation processor. Now the plugin tries to translate these folders. 

1. Why does ``translateJ2objcLibs``not work?
2. Is there a way I can exclude '**/*processor*' java source folders from being translated?

**Comment by advayDev1** [comment]
I would strongly recommend you to be much more specific in your bug reports.
Also, you need to file *separate* bugs for each issue.
None of the above error messages have anything to do with the issue you filed earlier.

So, please file a separate bug for translateJ2objcLibs.  And yet another bug for the processor source code.  When you do, please make available more context.  For example:
- All of your projects' build.gradle and settings.gradle (redacted as you need)
- Exact commit that you are sync-ed to and built the plugin at.
- Layout of your projects (by directory)

I have 2 hunches about the translateJ2objcLibs issue, but the actual problem very much depends on how you setup your projects.
Of course, share only what you are comfortable, for example, by redacting information, and then possibly making gists.

It takes up a lot more time for both of us if you don't provide context.

**Comment by confile** [comment]
@advayDev1 Done see: https://github.com/j2objc-contrib/j2objc-gradle/issues/259 and https://github.com/j2objc-contrib/j2objc-gradle/issues/258

**Comment by brunobowden** [comment]
Now that #258 and #259 are both closed, is this still an issue?
